connd/qconnectionagent.cpp

```diff
@@ -423,30 +425,49 @@ void QConnectionAgent::setup()
         scanTimeoutInterval = confFile.value("scanTimerInterval", "1").toUInt(); //in minutes
 
         if (isStateOnline(netman->state())) {
-            qDebug() << "default route type:" << netman->defaultRoute()->type();
+            qCInfo(connAgent) << "Default route type:" << netman->defaultRoute()->type();
             if (netman->defaultRoute()->type() == "ethernet")
                 isEthernet = true;
             if (netman->defaultRoute()->type() == "cellular" && scanTimeoutInterval != 0)
                 scanTimer->start(scanTimeoutInterval * 60 * 1000);
 
         }
-        qDebug() << "config file says" << confFile.value("connected", "online").toString();
+        btTetheringEnabled = confFile.value("tetheringBtEnabled", false).toBool();
+        if (btTetheringEnabled) {
+            tetheringBtTech = netman->getTechnology("bluetooth");
+            if (tetheringBtTech) {
+                connect(tetheringBtTech, &NetworkTechnology::poweredChanged, this, &QConnectionAgent::technologyPowerChanged);
+                connect(tetheringBtTech, &NetworkTechnology::tetheringChanged,
+                                this, &QConnectionAgent::techTetheringChanged, Qt::UniqueConnection);
+                if (tetheringBtTech->powered()) {
+                    startTethering("bluetooth");
```

Comment:
Yes that'd be cleaner. I think btTetheringEnabled should already be set, but I think it's fine to reduce this to just tetheringBtTech->setTethering(true). This would also avoid Bluetooth being automatically enabled at boot if it wasn't already on (as in your comment above) which wasn't intended. When you turn tethering on, bluetooth gets turned on, and that's the only time that should happen.